### PR TITLE
 wrap the server io to support different encoding 

### LINKED ISSUE #225: Need a wrapper for the current server io to support different protocol and encoding in future
**What would you like to be added*:
To provide a wrapper for the server io to support the future extension. Also, returning the transportation data size is needed for the flow control purpose.

**Why is this needed**:
The communication mechanism between cloud and edge is websocket with json encoding. However, this cloud  be change to other protocol like quic with protobuf.

### PR DESCRIPTION
**Pull request by qizha**
**What type of PR is this?**

/kind feature

**What this PR does / why we need it**:
The communication mechanism between cloud and edge is websocket with json encoding. However, this cloud be change to other protocol like quic with protobuf.

**Which issue(s) this PR fixes**:

Fixes #225 

**Special notes for your reviewer**:

### CODE REVIEW

File: cloud/edgecontroller/pkg/cloudhub/common/io/hubio.go [hunk 1]
```diff
@@ -0,0 +1,51 @@
+package io
+
+import (
+	"encoding/json"
+	"github.com/gorilla/websocket"
```
Comment (sids-b) [inline cloud/edgecontroller/pkg/cloudhub/common/io/hubio.go]: nit: Please seperate, golang and vendor packages.

Comment (qizha) [reply cloud/edgecontroller/pkg/cloudhub/common/io/hubio.go]: code updated

**Comment by rohitsardesai83** [comment]
created an issue for the failing test : https://github.com/kubeedge/kubeedge/issues/231

**Pull request merged**